src/as-translation.c

```diff
@@ -194,6 +232,8 @@ as_translation_load_from_xml (AsTranslation *tr, AsContext *ctx, xmlNode *node,
 	if (priv->kind == AS_TRANSLATION_KIND_UNKNOWN)
 		return FALSE;
 
+	as_translation_set_source_locale (tr, (gchar*) xmlGetProp (node, (xmlChar*) "source_locale"));
```

Comment:
Please use `as_xml_get_prop_value (node, "source_locale")` for new code (just a convenience macro, but it improves readability and makes refactoring easier).
You are also leaking memory here, as the result of `xmlGetProp` needs to be freed.

You can solve all of this with just a few convenience functions:
```c
as_ref_string_assign_transfer (&priv->source_locale,
                               as_xml_get_prop_value_refstr (node, "source_locale"));
```

Reply:
Oops, thanks for the suggestion